packages/react/src/reconciler.ts

```diff
@@ -27,148 +30,166 @@ export type Reconciler = ReactReconciler<
   PublicInstance
 >;
 
-export const reconciler = reactReconciler<
-  Type,
-  Props,
-  RemoteRoot,
-  ViewInstance,
-  TextInstance,
-  SuspenseInstance,
-  HydratableInstance,
-  PublicInstance,
-  HostContext,
-  UpdatePayload,
-  ChildSet,
-  TimeoutHandle,
-  NoTimeout
->({
-  now: Date.now,
-
-  // Timeout
-  scheduleTimeout: setTimeout,
-  cancelTimeout: clearTimeout,
-  noTimeout: false,
-  // @see https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMHostConfig.js#L408
-  queueMicrotask: (callback) =>
-    Promise.resolve(null).then(callback).catch(handleErrorInNextTick),
-
-  isPrimaryRenderer: true,
-  supportsMutation: true,
-  supportsHydration: false,
-  supportsPersistence: false,
-
-  // Context
-  getRootHostContext() {
-    return {};
-  },
-  getChildHostContext(context) {
-    return context;
-  },
-
-  // Instances
-  createTextInstance(text, root) {
-    return root.createText(text);
-  },
-  createInstance(type, allProps, root) {
-    const {children: _children, ...props} = allProps;
-    return root.createComponent(type, props);
-  },
-
-  // Updates
-  commitTextUpdate(text, _oldText, newText) {
-    text.updateText(newText);
-  },
-  prepareUpdate(_instance, _type, oldProps, newProps) {
-    const updateProps: Record<string, unknown> = {};
-    let needsUpdate = false;
-
-    for (const key in oldProps) {
-      if (!has(oldProps, key) || key === 'children') {
-        continue;
+export const createReconciler = (
+  hostConfig?: HostConfig<
```

Comment:
Can we make this a more specific options object, that is just `options?: {primary?: boolean}`? I don't think we need to leave space for overriding the rest of the options here.